src/hooks/useNewsApi.ts

```diff
@@ -36,37 +37,38 @@ const useNewsApi = () => {
     [navigate],
   );
 
-  const fetchNewsApi = useCallback(async () => {
-    setLoading(true);
+  const fetchNewsApi = useCallback(
+    async (signal: AbortSignal) => {
+      setLoading(true);
 
-    const shouldCreateNewInstance = api.current.apiSlug !== source;
-    const apiToUse = shouldCreateNewInstance ? ApiFactory.createApi(source) : api.current;
+      const shouldCreateNewInstance = api.current.apiSlug !== source;
+      const apiToUse = shouldCreateNewInstance ? ApiFactory.createApi(source) : api.current;
 
-    if (!search) {
-      handleEmptySearchParams(apiToUse, state);
+      if (!search) {
+        handleEmptySearchParams(apiToUse, state);
 
-      return;
-    }
+        return;
+      }
 
-    try {
-      await apiToUse.fetchNews(search);
-      setError(null);
-    } catch (err: unknown) {
-      setError(err as Error);
-    } finally {
-      if (shouldCreateNewInstance) api.current = apiToUse;
-      setLoading(false);
-    }
-  }, [source, search, handleEmptySearchParams, state]);
+      try {
+        await apiToUse.fetchNews(search, signal);
+        setError(null);
+      } catch (err: unknown) {
+        setError(err as Error);
+      } finally {
+        if (shouldCreateNewInstance) api.current = apiToUse;
+        setLoading(false);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [source, search, handleEmptySearchParams, state, trigger],
```

Comment:
I'm not sure if you need all of these as dependencies

Reply:
ESLint keeps warning me if I remove one of these dependencies, except trigger.

Reply:
Removed a lot of dependencies here after refactoring